src/ast/mod.rs

```diff
@@ -2479,19 +2482,116 @@ impl fmt::Display for Statement {
                 temporary,
                 if_not_exists,
                 name,
+                data_type,
+                sequence_options,
+                owned_by,
             } => {
+                let as_type: String = if let Some(dt) = data_type.as_ref() {
+                    //Cannot use format!(" AS {}", dt), due to format! is not available in --target thumbv6m-none-eabi
+                    // " AS ".to_owned() + &dt.to_string()
+                    [" AS ", &dt.to_string()].concat()
+                } else {
+                    "".to_string()
+                };
                 write!(
                     f,
-                    "CREATE {temporary}SEQUENCE {if_not_exists}{name}",
+                    "CREATE {temporary}SEQUENCE {if_not_exists}{name}{as_type}",
                     if_not_exists = if *if_not_exists { "IF NOT EXISTS " } else { "" },
                     temporary = if *temporary { "TEMPORARY " } else { "" },
-                    name = name
+                    name = name,
+                    as_type = as_type
+                )?;
+                for sequence_option in sequence_options {
+                    write!(f, "{}", sequence_option)?;
+                }
+                if let Some(ob) = owned_by.as_ref() {
+                    write!(f, " OWNED BY {}", ob)?;
+                }
+                write!(f, "")
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+/// Can use to describe options in create sequence or table column type identity
```

Comment:
This looks consistent with postgres 👍 

https://www.postgresql.org/docs/15/sql-createsequence.html